hack/jenkins/build.sh

```diff
@@ -34,15 +34,14 @@ export PATH=$PATH:/usr/local/go/bin
 export KUBE_RELEASE_RUN_TESTS=n
 export KUBE_SKIP_CONFIRMATIONS=y
 
-# Clean stuff out. Assume the worst - the last build may have left the
-# tree in an odd state. There's a Jenkins git clean plugin, but we
-# have the docker images to worry about as well, so be really pedantic
-# about cleaning.
+# Clean stuff out. Assume the last build left the tree in an odd
+# state, but do assume that Docker is capable of managing its
+# tags/images correctly.
 rm -rf ~/.kube*
 make clean
 git clean -fdx
-docker ps -aq | xargs -r docker rm
-docker images -q | xargs -r docker rmi
+# docker ps -aq | xargs -r docker rm
+# docker images -q | xargs -r docker rmi
```

Comment:
Add a comment to explain what the commented out lines do, and why someone might want to uncomment them.
Other than that, LGTM.